packages/browser/__tests__/login/oidc/redirectHandler/AuthCodeRedirectHandler.spec.ts

```diff
@@ -281,37 +310,56 @@ describe("AuthCodeRedirectHandler", () => {
     // We use ts-ignore comments here only to access mock call arguments
     /* eslint-disable @typescript-eslint/ban-ts-comment */
     it("returns an authenticated bearer fetch by default", async () => {
+      /* tslint:disable-next-line:no-any */
+      (global as any).localStorage = new LocalStorageMock();
+
       mockFetch(
         new Response("", {
           status: 200,
         })
       );
 
+      const testIssuer = "some test Issuer";
       const authCodeRedirectHandler = getAuthCodeRedirectHandler({
         storageUtility: mockStorageUtility({
+          "solidClientAuthenticationUser:mySession": {
+            issuer: testIssuer,
+          },
           "solidClientAuthenticationUser:oauth2_state_value": {
             sessionId: "mySession",
           },
         }),
       });
+
       const redirectInfo = await authCodeRedirectHandler.handle(
         "https://coolsite.com/?code=someCode&state=oauth2_state_value"
       );
+
       // This will call the oidc-client-ext module, which is mocked at
       // the top of this file. The purpose of this test is to check that, if
       // no additional information is given, the `getBearerToken` method is called
       // (as opposed to the getDpopToken one), which here returns a mock token
       // with the value "some token".
       await redirectInfo.fetch("https://some.other.url");
+
       // @ts-ignore
       const header = window.fetch.mock.calls[0][1].headers.Authorization;
       expect(
         // @ts-ignore
         header
       ).toMatch(/^Bearer some token$/);
+
+      // Also check that our issuer was stored correctly (specifically in
+      // 'localStorage').
+      expect(window.localStorage.getItem(KEY_CURRENT_ISSUER)).toEqual(
+        testIssuer
+      );
```

Comment:
Shouldn't the `handle` method always store the issuer, regardless of the type of token ? In this case, I think an independent test would be best.

Reply:
Yeah, I did consider it - and I'm kinda 50-50 on it. It was just 1-point ticket, and so I lumped it into existing tests. My justification was just that there's no logic involved now (at least not now that the 'bearer token' tests have been updated to include `issuer` extraction, which is actually an unrelated improvement!), so the fact we're testing for this behaviour on existing happy-path tests is kinda Ok I think.